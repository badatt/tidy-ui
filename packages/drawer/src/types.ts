import { HTMLAttributes } from 'react';
import { IBaseProps } from '@tidy-ui/types';

/**
 * Drawer props
 */
export interface IDrawerProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Direction of the tooltip
   *
   * @type {'right' | 'left'}
   * @default 'left'
   */
  anchor?: 'right' | 'left';

  /**
   * If `true`, opens the drawer
   *
   * @default false
   */
  isOpen: boolean;

  /**
   * Callback when the drawer is closed
   *
   * @type {?((e?: any | undefined) => void)}
   */
  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
  onClose: (e?: any) => void;
}
