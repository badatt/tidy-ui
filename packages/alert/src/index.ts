export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
