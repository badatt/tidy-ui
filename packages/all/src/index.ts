export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
