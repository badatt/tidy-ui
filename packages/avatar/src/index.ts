export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
