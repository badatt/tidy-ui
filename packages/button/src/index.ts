export * from './Button';
export * from './Anchor';
