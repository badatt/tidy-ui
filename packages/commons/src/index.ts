export * from './actions';
export * from './hooks';
export * from './icons';
export * from './provider';
export * from './reducers';
export * from './theme';
export * from './utils';
