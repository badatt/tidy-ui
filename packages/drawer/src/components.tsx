import { applyStandardOverrideStyles, color, css, hsla, styled } from '@tidy-ui/commons';
import { IDrawerProps } from './types';

/**
 * Internal Drawer root component
 *
 * @internal
 */
const DrawerRoot = styled.div<Pick<IDrawerProps, 'anchor'>>`
  z-index: ${Number.MAX_SAFE_INTEGER};
  position: fixed;
  background-color: ${hsla(color.slate[900], 0.7)};
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  ${({ anchor }) => css`
    flex-direction: ${anchor === 'left' ? 'row' : 'row-reverse'};
  `}
`;

/**
 * Internal Drawer content component
 *
 * @internal
 */
const DrawerContent = styled.div`
  display: flex;
  flex-direction: column;
  ${({ theme: { palette } }) => css`
    height: 100vh;
    background-color: ${palette.background.default};
  `}
  ${applyStandardOverrideStyles}
`;

/**
 * Internal Drawer close section component
 *
 * @internal
 */
const CloseSection = styled.div<Pick<IDrawerProps, 'anchor'>>`
  display: flex;
  ${({ anchor }) => css`
    flex-direction: ${anchor === 'left' ? 'row-reverse' : 'row'};
  `}
`;

/**
 * Internal close button component
 *
 * @internal
 */
const CloseButton = styled.button`
  padding: 0.5rem;
  ${({ theme: { palette } }) => css`
    color: ${palette.text.primary};
  `}
`;

export { CloseButton, CloseSection, DrawerContent, DrawerRoot };
