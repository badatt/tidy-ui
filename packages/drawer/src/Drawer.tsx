import React from 'react';
import { Icon, useIsMounted, usePortal } from '@tidy-ui/commons';
import { CloseButton, CloseSection, DrawerContent, DrawerRoot } from './components';
import { IDrawerProps } from './types';

/**
 * Drawer can be used for menu section in your app.
 * You can use Drawer to launch a portal
 * emerge from left/right side of the page
 */
const Drawer = React.forwardRef<HTMLDivElement, IDrawerProps>((props, ref) => {
  const { children, isOpen, onClose, ...rest } = props;

  const isMounted = useIsMounted();
  const { Portal } = usePortal();

  const handleClose = React.useCallback(
    (e: React.MouseEvent<HTMLElement>) => {
      e.stopPropagation();
      if (isMounted()) {
        onClose?.(e);
      }
    },
    [isMounted, onClose],
  );

  if (!isOpen) {
    return null;
  }

  return (
    <Portal>
      <DrawerRoot ref={ref} role="navigation" {...rest}>
        <DrawerContent {...rest}>
          <CloseSection {...rest}>
            <CloseButton onClick={handleClose} role="button">
              <Icon ele={<Icon.Close />} />
            </CloseButton>
          </CloseSection>
          {children}
        </DrawerContent>
      </DrawerRoot>
    </Portal>
  );
});

Drawer.defaultProps = {
  anchor: 'left',
};

Drawer.displayName = 'Drawer';

export { Drawer };
