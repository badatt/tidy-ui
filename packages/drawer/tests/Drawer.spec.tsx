/**
 * @jest-environment jsdom
 */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Drawer } from '../src';

describe('Drawer', () => {
  it('Basic render', () => {
    const mockSetOpen = jest.fn();
    const mockSetClose = jest.fn();
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <div onClick={mockSetOpen} data-testid="open-btn">
          Open
        </div>
        <Drawer isOpen onClose={mockSetClose}>
          menu
        </Drawer>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
    const openBtn = tree.getByTestId('open-btn');
    fireEvent.click(openBtn);
    expect(mockSetOpen).toHaveBeenCalledTimes(1);
    const closeBtn = tree.getByRole('button');
    fireEvent.click(closeBtn);
    expect(mockSetClose).toHaveBeenCalledTimes(1);
  });

  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <button data-testid="open-btn">Open</button>
        <Drawer isOpen>menu</Drawer>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Drawer on right', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <button data-testid="open-btn">Open</button>
        <Drawer isOpen anchor="right">
          menu
        </Drawer>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Drawer is in closed state', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <button data-testid="open-btn">Open</button>
        <Drawer isOpen={false}>menu</Drawer>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('No close handler', () => {
    const mockSetOpen = jest.fn();
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <div onClick={mockSetOpen} data-testid="open-btn">
          Open
        </div>
        <Drawer isOpen={true}>menu</Drawer>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
    const openBtn = tree.getByTestId('open-btn');
    fireEvent.click(openBtn);
    expect(mockSetOpen).toHaveBeenCalledTimes(1);
    const closeBtn = tree.getByRole('button');
    fireEvent.click(closeBtn);
  });
});
