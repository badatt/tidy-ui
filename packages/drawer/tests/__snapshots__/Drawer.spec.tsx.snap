// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  background-color: hsla(216,56%,98%,1);
}

.c2 {
  padding: 0.5rem;
  position: absolute;
  z-index: 1;
  background-color: hsla(215,25%,27%,0.1);
  color: hsla(217,33%,17%,1);
  top: 0;
  right: 0;
}

.c2:hover {
  background-color: hsla(215,25%,27%,0.2);
}

<body>
    <div>
      <div
        data-testid="open-btn"
      >
        Open
      </div>
    </div>
    <div
      class="c0"
      role="navigation"
    >
      <div
        class="c1"
      >
        <button
          class="c2"
          role="button"
        >
          <span
            style="display: flex; height: 1.25em; width: 1.25em;"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              style="display: flex; height: 1.25em; width: 1.25em;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        menu
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="open-btn"
    >
      Open
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drawer Custom element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  background-color: hsla(215,15%,16%,1);
}

.c2 {
  padding: 0.5rem;
  position: absolute;
  z-index: 1;
  background-color: hsla(215,25%,27%,0.1);
  color: hsla(210,19%,73%,1);
  top: 0;
  right: 0;
}

.c2:hover {
  background-color: hsla(215,25%,27%,0.2);
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Open
      </button>
    </div>
    <div
      class="c0"
      role="navigation"
    >
      <div
        class="c1"
      >
        <button
          class="c2"
          role="button"
        >
          <span
            style="display: flex; height: 1.25em; width: 1.25em;"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              style="display: flex; height: 1.25em; width: 1.25em;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        <div>
          menu
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Open
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drawer Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  background-color: hsla(215,15%,16%,1);
}

.c2 {
  padding: 0.5rem;
  position: absolute;
  z-index: 1;
  background-color: hsla(215,25%,27%,0.1);
  color: hsla(210,19%,73%,1);
  top: 0;
  right: 0;
}

.c2:hover {
  background-color: hsla(215,25%,27%,0.2);
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Open
      </button>
    </div>
    <div
      class="c0"
      role="navigation"
    >
      <div
        class="c1"
      >
        <button
          class="c2"
          role="button"
        >
          <span
            style="display: flex; height: 1.25em; width: 1.25em;"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              style="display: flex; height: 1.25em; width: 1.25em;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        menu
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Open
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drawer Drawer is in closed state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="open-btn"
      >
        Open
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Open
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drawer Drawer on right 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  z-index: 9007199254740991;
  position: fixed;
  background-color: hsla(222,47%,11%,0.7);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  background-color: hsla(216,56%,98%,1);
}

.c2 {
  padding: 0.5rem;
  position: absolute;
  z-index: 1;
  background-color: hsla(215,25%,27%,0.1);
  color: hsla(217,33%,17%,1);
  top: 0;
  left: 0;
}

.c2:hover {
  background-color: hsla(215,25%,27%,0.2);
}

<body>
    <div>
      <button
        data-testid="open-btn"
      >
        Open
      </button>
    </div>
    <div
      class="c0"
      role="navigation"
    >
      <div
        class="c1"
      >
        <button
          class="c2"
          role="button"
        >
          <span
            style="display: flex; height: 1.25em; width: 1.25em;"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              style="display: flex; height: 1.25em; width: 1.25em;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        menu
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="open-btn"
    >
      Open
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
